EveHQ.Prism/Classes/StructureBonus.vb

```diff
@@ -0,0 +1,229 @@
+''' </summary>
+Public Class StructureBonus
+
+    Private _group As String
+    Private _materialEfficiency As Decimal
+    Private _timeEfficiency As Decimal
+    Private _costEfficiency As Decimal
+    Private _scienceBonus As Boolean
+
+    ''' <summary>
+    ''' The Group to which the Bonus should be Applied
+    ''' </summary>
+    Public Property Group As String
+        Get
+            Return _group
+        End Get
+        Set(value As String)
+            _group = value
+        End Set
+    End Property
+
+    ''' <summary>
+    ''' The Material Efficiency Rate provided Bonus
+    ''' </summary>
+    Public Property MaterialEfficiency As Decimal
+        Get
+            Return _materialEfficiency
+        End Get
+        Set(value As Decimal)
+            _materialEfficiency = value
+        End Set
+    End Property
+
+    ''' <summary>
+    ''' The Time Efficiency Rate provided Bonus
+    ''' </summary>
+    Public Property TimeEfficiency As Decimal
+        Get
+            Return _timeEfficiency
+        End Get
+        Set(value As Decimal)
+            _timeEfficiency = value
+        End Set
+    End Property
+
+    ''' <summary>
+    ''' The Cost Efficiency Rate provided Bonus
+    ''' </summary>
+    Public Property CostEfficiency As Decimal
+        Get
+            Return _costEfficiency
+        End Get
+        Set(value As Decimal)
+            _costEfficiency = value
+        End Set
+    End Property
+
+    ''' <summary>
+    ''' Determine if the Bonus should be applied to a Science job or a production job
+    ''' </summary>
+    Public Property IsScienceBonus As Boolean
+        Get
+            Return _scienceBonus
+        End Get
+        Set(value As Boolean)
+            _scienceBonus = value
+        End Set
+    End Property
+
+    Private _appliesToMarketGroups As Dictionary(Of Integer, Integer)
+
+    ''' <summary>
+    ''' Determine to which Item Market Group ID the bonus is applied
+    ''' http://content.eveonline.com/www/newssystem/media/70592/1/EngineeringRigsv2.jpg
+    ''' 
+    ''' Key indicate item market group ID
+    ''' Value indicate tech level, 0 mean no restriction
+    ''' </summary>
+    Public ReadOnly Property AppliesToMarketGroups() As Dictionary(Of Integer, Integer)
+        Get
+            Return _appliesToMarketGroups
+        End Get
+    End Property
+
+    ''' <summary>
+    ''' Search in the bonus array if we found the EveType Market Group ID for the associated Tech Level
+    ''' </summary>
+    ''' <param name="type">The EveType which should be checked</param>
+    ''' <returns></returns>
+    Public Function IsAppliedTo(type As EveType) As Boolean
+
+        Return (From group In Me._appliesToMarketGroups
+                Where group.Key = type.MarketGroupId
+                Where StaticData.GetAttributeDataForItem(type.Id).  ' Fetch item attributes and check tech level (422)
+                    Where(Function(a) a.Key.Equals(422)).
+                    First().
```

Comment:
It can rise an Exception if there is no such itemType. May be in this place you should detect it and return false?